ui/src/components/FormModifications/FormModifications.test.tsx

```diff
@@ -0,0 +1,213 @@
+    ) => {
+        expect(input).toHaveAttribute('value', mods?.value);
+
+        invariant(typeof mods?.help === 'string');
+        expect(parentElement.textContent?.includes(mods.help)).toEqual(true);
+
+        invariant(typeof mods?.label === 'string');
+        expect(parentElement.textContent?.includes(mods.label)).toEqual(true);
+    };
+
+    expect(componentInput).toHaveAttribute('disabled');
+    verifyAllProps(componentParentElement, componentInput, mods1Field1);
+
+    expect(component2Input).toHaveAttribute('disabled');
+    verifyAllProps(component2ParentElement, component2Input, mods1Field2);
+
+    await userEvent.type(componentMakingModsTextBox1, secondValueToInput);
+
+    expect(componentInput).not.toHaveAttribute('disabled');
+    verifyAllProps(componentParentElement, componentInput, mods2Field1);
+
+    expect(component2Input).not.toHaveAttribute('disabled');
+    verifyAllProps(component2ParentElement, component2Input, mods2Field2);
+});
+
+it('verify markdown modifications', async () => {
+    setUpConfigWithDefaultValue();
+    renderModalWithProps(props);
+
+    const firstValueToInput = 'a';
+    const secondValueToInput = 'aa';
+
+    const [componentParentElement] = getAndVerifyHTMLTextComponentsForField(
+        firstStandardTextField.field
+    );
+    const [component2ParentElement] = getAndVerifyHTMLTextComponentsForField(
+        secondStandardTextField.field
+    );
+
+    const componentMakingModsTextBox1 = getAndVerifyHTMLTextComponentsForField(
+        secondModificationField.field
+    )[1];
+
+    const firstElementOuterHTMLBeforeMods = componentParentElement?.outerHTML;
+
+    const mods1Field2 = findMods(
+        secondModificationField,
+        firstValueToInput,
+        secondStandardTextField.field
+    );
+
+    const mods2Field1 = findMods(
+        secondModificationField,
+        secondValueToInput,
+        firstStandardTextField.field
+    );
+    const mods2Field2 = findMods(
+        secondModificationField,
+        secondValueToInput,
+        secondStandardTextField.field
+    );
+
+    await userEvent.type(componentMakingModsTextBox1, firstValueToInput);
+
+    const firstElementOuterHTMLAfterMods = componentParentElement?.outerHTML;
+
+    // only id specified as modification so no changes should occure
+    expect(firstElementOuterHTMLBeforeMods).toEqual(firstElementOuterHTMLAfterMods);
+
+    expect(component2ParentElement.textContent).toMatchInlineSnapshot(
+        `"Standard text label second fieldmarkdown message to open conf page and explain sthStandard Text help second field"`
+    );
+
+    const anchorElementField2 = component2ParentElement?.querySelector('a');
+    expect(anchorElementField2?.getAttribute('href')).toEqual(
+        mods1Field2?.markdownMessage?.markdownType === 'hybrid' &&
+            mods1Field2?.markdownMessage?.link
+    );
+
+    expect(anchorElementField2?.textContent).toMatchInlineSnapshot(`"conf page"`);
```

Comment:
we want to check the link with a text and attribute:
```ts
    const a = within(component2ParentElement).getByRole('link', { name: 'conf page' });
    expect(a).toHaveAttribute('href', mods1Field2.markdownMessage.link);
```

Reply:
done